src/components/Confirm/index.js

```diff
@@ -172,12 +150,33 @@ class Confirm extends Component {
     }
   }
 
+  sendMultiframeSelfie = (snapshot, selfie, token, url, onSuccess, onError) => {
+    const snapshotData = {
+      file: {
+        blob: snapshot.blob,
+        filename: snapshot.filename
+      }
+    }
+    const { blob, filename, sdkMetadata } = selfie
+  
+    new Promise((resolve, reject) => {
+      uploadSnapshot(snapshotData, url, token, resolve, reject)
+    })
+    .then((res) => {
+      uploadLivePhoto({ file: { blob, filename }, sdkMetadata, snapshots: [res.id]}, url, token, onSuccess, onError)
+    })
+    .catch(() => {
+      this.setState({ uploadInProgress: false })
+      this.setError('SERVER_ERROR')
+    })
+  }
+
   handleSelfieUpload = ({ snapshot, ...selfie }, token) => {
     const url = this.props.urls.onfido_api_url
     // if snapshot is present, it needs to be uploaded together with the user initiated selfie
     if (snapshot) {
       sendEvent('Starting multiframe selfie upload')
-      chainMultiframeUpload(snapshot, selfie, token, url, this.onApiSuccess, this.onApiError)
+      this.sendMultiframeSelfie(snapshot, selfie, token, url, this.onApiSuccess, this.onApiError)
```

Comment:
LGTM, but just out of my curiosity:
Is the `this` really necessary?

Reply:
Yes, `sendMultiframeSelfie` is a function that belongs to our `Confirm` component class. If you call it without the `this` you would get an error saying that `sendMultiframeSelfie` is not a function because `handleSelfieUpload` doesn't know about `sendMultiframeSelfie` unless you specify the context